controllers/healthcheck_controller.go

```diff
@@ -138,25 +139,35 @@ func (r *HealthCheckReconciler) processHealthCheck(ctx context.Context, req ctrl
 	if hcSpec.Workflow.Resource != nil {
 		wfNamePrefix := hcSpec.Workflow.GenerateName
 		wfNamespace := hcSpec.Workflow.Resource.Namespace
-		repeatAfterSec := hcSpec.RepeatAfterSec
 		level := hcSpec.Level
 		sa := hcSpec.Workflow.Resource.ServiceAccount
 		now := metav1.Time{Time: time.Now()}
 		var finishedAtTime int64
 		if healthCheck.Status.FinishedAt != nil {
 			finishedAtTime = healthCheck.Status.FinishedAt.Time.Unix()
 		}
+
 		// workflows can be paused by setting repeatAfterSec to <= 0.
-		if repeatAfterSec <= 0 {
-			log.Info("Workflow will be skipped due to repeatAfterSec value", "repeatAfterSec", repeatAfterSec)
+		if hcSpec.RepeatAfterSec <= 0 && healthCheck.Spec.Scheduler.Cron == "" {
+			log.Info("Workflow will be skipped due to repeatAfterSec value", "repeatAfterSec", hcSpec.RepeatAfterSec)
 			healthCheck.Status.Status = "Stopped"
-			healthCheck.Status.ErrorMessage = fmt.Sprintf("workflow execution is stopped due to spec.repeatAfterSec set to %d", repeatAfterSec)
+			healthCheck.Status.ErrorMessage = fmt.Sprintf("workflow execution is stopped due to spec.repeatAfterSec set to %d", hcSpec.RepeatAfterSec)
 			healthCheck.Status.FinishedAt = &now
 			return ctrl.Result{}, nil
-		} else if int(time.Now().Unix()-finishedAtTime) < repeatAfterSec {
+		} else if hcSpec.RepeatAfterSec <= 0 && healthCheck.Spec.Scheduler.Cron != "" {
+			scheduler, err := cron.ParseStandard(healthCheck.Spec.Scheduler.Cron)
+			if err != nil {
+				log.Error(err, "fail to parse cron")
+			}
+			// The value from scheduler next and substracting from current time is in fraction as we convert to int it will be 1 less than
+			// the intended reschedule so we need to add 1sec to get the actual value
+			healthCheck.Spec.RepeatAfterSec = int(scheduler.Next(time.Now()).Sub(time.Now())/time.Second) + 1
```

Comment:
```suggestion
			hcSpec.RepeatAfterSec = int(scheduler.Next(time.Now()).Sub(time.Now())/time.Second) + 1
```

double check to ensure that assigning back to the shadow var would work as expected; ignore this comment and part of the next one if it doesn't work

Reply:
This will not work as hcSpec.RepeatAfterSec is updating the local var not the spec.
So in this case we need to use: healthCheck.Spec.RepeatAfterSec